src/renderer/routes/flow_chart/components/node-edit-menu/SecretSelect.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/renderer/components/ui/select";
+import { baseClient } from "@/renderer/lib/base-client";
+import { useState, useEffect } from "react";
+
+export type SelectProps = {
+  onValueChange: (value: string) => void;
+  value: string | number | boolean | null | undefined;
+};
+
+export const SecretSelect = ({ onValueChange, value }: SelectProps) => {
+  const [secrets, setSecrets] = useState<string[]>([]);
+  const [found, setFound] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fetchCredentials = async () => {
+      try {
+        const res = await baseClient.get("env");
+        const keys = res.data.map((d) => d.key);
+        setSecrets(keys);
+        setFound(keys.length > 0);
+        if (found) {
+          value ??= keys[0];
+        }
```

Comment:
No need to do this, to actually change some state you need to use the setter and not assignment